### PR TITLE
Implemented `std::iter::Sum` for `FieldElement`

### LINKED ISSUE #63: Implement `std::iter::Sum` for `FieldElement`
The task is to implement [this trait](https://doc.rust-lang.org/std/iter/trait.Sum.html) for [`FieldElement`](https://github.com/lambdaclass/lambdaworks/blob/main/math/src/field/element.rs#L8).

### PR DESCRIPTION
**Pull request by AleksanderMisztal**
Resolves #63

### CODE REVIEW

File: math/src/field/element.rs [hunk 1]
```diff
@@ -103,6 +104,16 @@ where
     }
 }
 
+/// Sum operator for field elements
+impl<F> Sum<FieldElement<F>> for FieldElement<F>
+where
+    F: IsField,
+{
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Self::zero(), |a, b| a + b)
```
Comment (ilitteri) [inline math/src/field/element.rs:113]: Could we change the parameters name to `addend` and `augend` like this?
```suggestion
        iter.fold(Self::zero(), |augend, addend| augend + addend)
```

**Review by ilitteri — CHANGES_REQUESTED** [review]
The PR looks good! Just one minor change and we can merge!

**Review by ilitteri — APPROVED** [review]
LGTM

**Comment by AleksanderMisztal** [comment]
Btw I have some spare time over the next few days, I could tackle something bigger. What's the best way to message you to discuss?

**Comment by ilitteri** [comment]
> Btw I have some spare time over the next few days, I could tackle something bigger. What's the best way to message you to discuss?

First of all thanks for addressing the issue.

I think it'd be awesome if you tackle something bigger.

Regarding your question, I'll @unbalancedparentheses and @MauroToscano answer it.

**Review by jrchatruc — APPROVED** [review]

**Comment by unbalancedparentheses** [comment]
Hi! The best way is that you join the telegram and we talk over there. You can find the telegram group in the readme!

**Comment by AleksanderMisztal** [comment]
Thanks! But looks like that link is broken 👀

**Comment by ilitteri** [comment]
> Thanks! But looks like that link is broken 👀

Fixed!

**Comment by ilitteri** [comment]
Could you merge from `main` so we can run the CI?

**Comment by ilitteri** [comment]
Sorry for the dealy, just one more detail. Could we add a test for this?

**Comment by MauroToscano** [comment]
There is a small conflict, if it's fix we can merge it

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**